tests/yast2_gui/yast2_firewall_libyui.pm

```diff
@@ -0,0 +1,207 @@
+    select_console 'x11', await_console => 0;
+    YaST::Module::open(module => 'firewall', ui => 'qt');
+    assert_screen 'yast2_firewall_start-up';
+    $testapi::distri->get_firewall()->get_main_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->get_start_up_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->stop_firewall();
+    save_screenshot;
+    $testapi::distri->get_firewall()->accept_change();
+    assert_screen 'generic-desktop';
+    select_console 'root-console';
+    assert_script_run("! (firewall-cmd --state) | grep 'not running'");
+}
+
+sub verify_service_started {
+
+    record_info('Start-Up', "Managing the firewalld service: Start");
+    select_console 'x11', await_console => 0;
+    YaST::Module::open(module => 'firewall', ui => 'qt');
+    assert_screen 'yast2_firewall_start-up';
+    $testapi::distri->get_firewall()->get_start_up_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->start_firewall();
+    save_screenshot;
+    $testapi::distri->get_firewall()->accept_change();
+    assert_screen 'generic-desktop';
+    select_console 'root-console';
+    assert_script_run "firewall-cmd --state | grep running";
+}
+
+sub verify_interface {
+    my (%args) = @_;
+    if ($testapi::distri->get_firewall()->verify_interface($args{device}, $args{zone}) != 1) {
+        die("Verify interface failed" . $args{device} . " " . $args{zone});
+    } else {
+        record_info('Verify interface ok', $args{device} . " " . $args{zone});
+    }
+}
+
+sub change_interface_zone {
+    my (%args) = @_;
+    $testapi::distri->get_firewall()->get_interfaces_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->set_interface_zone($args{device}, $args{zone});
+    save_screenshot;
+}
+
+sub verify_zone {
+    my (%args) = @_;
+
+    my $interfaces = $args{interfaces} //= 'no_interfaces';
+    my $default = $args{default} //= 'no_default';
+    my $menu_selected = $args{menu_selected} //= 0;
+    $testapi::distri->get_firewall()->get_zones_page();
+    save_screenshot;
+    if ($testapi::distri->get_firewall()->verify_zone($args{name}, $interfaces, $default) != 1) {
+        die("Verify zone failed! " . $args{name} . " " . $args{device} . " " . $interfaces . " " . $default);
+    } else {
+        record_info('verify zone ok! ', $args{name} . " " . $args{device} . " " . $interfaces . " " . $default);
+    }
+}
+
+sub set_default_zone {
+    my $zone = shift;
+    $testapi::distri->get_firewall()->get_zones_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->set_default_zone($zone);
+    save_screenshot;
+}
+
+sub configure_zone {
+    my (%args) = @_;
+
+    $testapi::distri->get_firewall()->select_zone_page($args{zone});
+    save_screenshot;
+    $testapi::distri->get_firewall()->add_service($args{zone}, $args{service});
+    $testapi::distri->get_firewall()->add_tcp_port($args{port});
+}
+
+sub configure_firewalld {
```

Comment:
Current control layer already wrap some operation to avoid too much detail shown in test code, it's not easy thing to balance between flexible api vs simple easy api.